### ISSUE TITLE
heat convection coeff if-else

### ISSUE THREAD

**Issue by PMehrfeld** [RWTH-EBC/AixLib#711]
**What is the problem?**

- When using calculation method acc. to EN ISO 6946 the model `AixLib.Utilities.HeatTransfer.HeatConv_inside` switches between to heat convection coefficients `alpha` depending on temperature at wall and in the room.

Equation is:
```Modelica
      if port_b.T >= port_a.T then
        alpha = 5;
        // downward heat flow
      else
        alpha = 0.7;
      end if;
```

- This leads to a hard state event forcing the solver to solve a totally changed equation system (e.g. heat flows turn around and value of heat transfer changes its magnitude).

**How do we want to solve it? Describe the solution you'd like**

- Maybe just use the [`smooth-function`](https://build.openmodelica.org/Documentation/ModelicaReference.Operators.%27smooth()%27.html). Or any other kind of make the zero-crossing differentiable.

**Describe alternatives you've considered**

An alternative would be to emphasize this issue in the documentation. 
Since the EN ISO approach is  correct and directly taken from the standard, I defnitly vote for let this calculation method in the model.

The approach acc. to Bernd Glueck is a differentiable function, so it should be recommended.

**Additional context**

The approach acc. to Bernd Glueck follows 2 functions (depending if the wall is a ceiling or a floor (or a vertical wall)). The two functions have the following behavior:
![image](https://user-images.githubusercontent.com/11769433/57176665-dfa8bb80-6e5a-11e9-9bc5-6d1935522c09.png)

**Comment by PMehrfeld** [comment]
In a comparison simulation (MFH for 1 day) but with heat convection inside according to Bernd Glueck, the following performance boost was achieved. The `alpha` alpha in the first par is totally different. 
The literature source should be checked again, whether it depends on cooling or heating and thus probably also on `port_a.T < port_b.T`.

![image](https://user-images.githubusercontent.com/11769433/57176971-4e881380-6e5f-11e9-9376-482b6807bba0.png)

**Comment by PMehrfeld** [comment]
Regarding literature source of Bern Glueck ([German PDF](http://berndglueck.de/dl/?dl=LowEx+Bericht_LowEx.pdf)) p. 26:
![image](https://user-images.githubusercontent.com/11769433/57177054-5e542780-6e60-11e9-9650-6b433cedbe07.png)


Thus, it also depends on the direction of the heat flow, which can change during the simulation.
Within the model `AixLib.Utilities.HeatTransfer.HeatConv_inside` it is implemented for EN ISO 6946, but not for approach acc. to Bernd Glueck.

Additionally, from my point of view, there is absolutely no need to distinguish between floor or ceiling, but only between vertical and horizontal wall. For horizontal wall during the simulation the direction of the heat flow determines which value (EN ISO 6946: 5 or 0.7; Glueck: 2 or 0.54 *(posDiff^0.31)) must be used.

**Comment by PMehrfeld** [comment]
One solution (with which I came up quickly) might be something like the following.
However, an estimation of the deviations between both calculation methods should be included (maybe even in the example model).

```Modelica
model HeatConv_inside
...
protected 
  Modelica.SIunits.Temp_C posDiff=noEvent(abs(port_b.T - port_a.T))
    "Positive temperature difference";
equation 

  /*
        port_b -> wall
        port_a -> air
  */

  // ++++++++++++++++EN ISO 6946 Appendix A >>Flat Surfaces<<++++++++++++++++
  if calcMethod == 1 then
    if surfaceOrientation == 2 then
      // upward heat flow
      //if port_b.T >= port_a.T then
        alpha = smooth(2, if port_b.T >= port_a.T then 5 else 0.7);//5;
        // downward heat flow
      //else
        //alpha = 0.7;
      //end if;
    elseif surfaceOrientation == 3 then
      // downward heat flow
      //if port_b.T >= port_a.T then
        alpha = smooth(2, if port_b.T >= port_a.T then 0.7 else 5);//0.7;
        // upward heat flow
      //else
        //alpha = 5;
      //end if;
    else
      alpha = 2.5;
    end if;
  // ++++++++++++++++Bernd Glueck++++++++++++++++
  elseif calcMethod == 2 then

    // top side of horizontal plate
// ------------------------------------------------------
  if surfaceOrientation == 2 then
      alpha = smooth(2, if port_b.T >= port_a.T then 2*(posDiff^0.31) else 0.54*(posDiff^0.31));//2*(posDiff^0.31);  // equation 1.27, page 26 (Bernd Glueck: Heizen und Kuehlen mit Niedrigexergie - Innovative Waermeuebertragung und Waermespeicherung (LowEx) 2008)

// down side of horizontal plate
// ------------------------------------------------------

  else
    if surfaceOrientation == 3 then
       alpha = smooth(2, if port_b.T < port_a.T then 2*(posDiff^0.31) else 0.54*(posDiff^0.31));//0.54*(posDiff^0.31);  //equation 1.28, page 26 (Bernd Glueck: Heizen und Kuehlen mit Niedrigexergie - Innovative Waermeuebertragung und Waermespeicherung (LowEx) 2008)

// vertical plate
//-------------------------------------------------
    else
      alpha = 1.6*(posDiff^0.3);  // equation 1.26 page 26 (Bernd Glueck: Heizen und Kuehlen mit Niedrigexergie - Innovative Waermeuebertragung und Waermespeicherung (LowEx) 2008)
    end if;
  end if;
  // ++++++++++++++++alpha_custom++++++++++++++++
  else
    // if calcMethod == 3 then
    alpha = alpha_custom;
    // end if;
  end if;

  port_a.Q_flow = alpha*A*(port_a.T - port_b.T);
...
end HeatConv_inside;
```

**Comment by PMehrfeld** [comment]
Two simulation to test the solution given above (Glueck: //1; EN ISO: //2) 

![image](https://user-images.githubusercontent.com/11769433/57178483-feb24800-6e70-11e9-8c99-d11e144b9d91.png)


Bad news: still a steep slope in CPUtime curve with EN ISO approach. 
Thus, latter calculation approach needs a function with zero-crossing.

**Comment by PMehrfeld** [comment]
Summary after discussion with @AnaConstantin:

- [ ] In the documentation: Clarify limitations of both approaches
- [ ] Use `smooth()` as above shown.
- [ ] Correct approach acc. to Bernd Glueck such that heat flux  (or dT) can turn around during the simulation (solution from above).
- [ ] Linearize or use stable function for EN ISO approach around `dT = 0 K +/- e.g. 1 K`

**Comment by AnaConstantin** [comment]
@PMehrfeld on limitations:
- the Glück reference is actually for thermally activated walls, and it combines free with forced convection. Because it is nicely continuous around zero I like it for simple walls as well. Furthermore the numbers are not very different from the implementation in Buildings. But it's something worth noting. Going by the Figures 1.14 -1.15 - 1.16 from the cited reference we can say it works for tsurface-tair from -10 K to +30 K, which should be enough for our applications. The length of the surface vary from 1 m to 3 m, which ...ok...
- Glück also offers an overview of other coefficients in table 1.1 (also using a DIN Norm), but again for mixt convection
- DIN EN ISO 6946 offers in Table C.1 the values we have in the model, however they are for surfaces "die an eine stark belüftete Luftschicht angrenzen" (which, NO). Digging deeper in the norm we get to chapter 6.9 to which this wonderful values relate to, and it is not the application we need.

**Comment by PMehrfeld** [comment]
This stays open, until a correct merge into IBPSA is performed.

**Comment by PMehrfeld** [comment]
@katharinabr : Thanks a lot for all this renaming and the revision of equations. Now the numerical calculations are more stable and faster.

Closed by #753
